src/osd/OSD.h

```diff
@@ -1113,6 +1113,11 @@ class OSDService {
   epoch_t boot_epoch;  // _first_ epoch we were marked up (after this process started)
   epoch_t up_epoch;    // _most_recent_ epoch we were marked up
   epoch_t bind_epoch;  // epoch we last did a bind to new ip:ports
+
+  // for PGs to request new osdmaps
+  std::mutex subscribed_epoch_lock;
+  epoch_t subscribed_epoch{0};
```

Comment:
Don't we already have a variable somewhere for the epoch we've requested from the monitor? It shouldn't get dumb located.

Reply:
not for the latest map. I'll move it to the osd from osdservice though